core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -0,0 +1,96 @@
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Task for validating security configurations.
+ */
+public final class SecureHdfsValidationTask extends HdfsValidationTask {
+  /**
+   * Regular expression to parse principal used by Alluxio to connect to secure
+   * HDFS.
+   *
+   * @see <a href="https://web.mit.edu/kerberos/krb5-1.5/krb5-1.5.4/doc/krb5-user/What-is-a-Kerberos-Principal_003f.html">Kerberos documentation</a>
+   * for more details.
+   */
+  private static final Pattern PRINCIPAL_PATTERN =
+      Pattern.compile("(?<primary>[\\w][\\w-]*\\$?)(/(?<instance>[\\w]+))?(@(?<realm>[\\w]+))?");
+
+  private static final Map<String, PropertyKey> PRINCIPAL_MAP = ImmutableMap.of(
+      "master", PropertyKey.MASTER_PRINCIPAL,
+      "worker", PropertyKey.WORKER_PRINCIPAL);
+  private static final Map<String, PropertyKey> KEYTAB_MAP = ImmutableMap.of(
+      "master", PropertyKey.MASTER_KEYTAB_KEY_FILE,
+      "worker", PropertyKey.WORKER_KEYTAB_FILE);
+
+  private final String mProcess;
+
+  /**
+   * Constructor of {@link SecureHdfsValidationTask}.
+   *
+   * @param process type of the process on behalf of which this validation task is run
+   */
+  public SecureHdfsValidationTask(String process) {
+    mProcess = process.toLowerCase();
+  }
+
+  @Override
+  public boolean validate(Map<String, String> optionsMap) {
+    if (!super.validate(optionsMap)) {
+      return false;
+    }
+    if (!validatePrincipalLogin()) {
+      System.err.format("Principal login test failed.");
+      return false;
+    }
+    return true;
+  }
+
+  private boolean validatePrincipalLogin() {
+    // Check whether can login with specified principal and keytab
+    String principal = Configuration.get(PRINCIPAL_MAP.get(mProcess));
+    String keytab = Configuration.get(KEYTAB_MAP.get(mProcess));
+    String primary;
+    String instance;
+    String realm;
+
+    Matcher matchPrincipal = PRINCIPAL_PATTERN.matcher(principal);
+    if (!matchPrincipal.matches()) {
+      System.err.format("Principal %s is not in the right format.%n", principal);
+      return false;
+    }
+    primary = matchPrincipal.group("primary");
+    instance = matchPrincipal.group("instance");
+    realm = matchPrincipal.group("realm");
+
+    // Login with principal and keytab
+    int exitVal =
+        Utils.getResultFromProcess(new String[] {"kinit", "-kt", keytab, principal}).getExitValue();
```

Comment:
is there a java library for the same?. and is krb5 libraries on the host OS a pre-requisite for running Alluxio (open-source) against secure HDFS

Reply:
Will look into this.

Reply:
Hadoop has a library for this, but current validation does not have this dependency. If we add this dependency, we need to account for multiple Hadoop versions. Similar to the previous comment.

Reply:
We discussed offline, and this may require us to re-arrange our entire validation-related code. Since this is not likely my recent priority, I suggest that we deal with this in a separate PR.